tests/test_analytics.py

```diff
@@ -8,4 +8,11 @@
 class TestAnalytics(unittest.TestCase):
 
     def setUp(self):
-        pass
\ No newline at end of file
+        pass
+    
+    def test_permutations(self):
+        self.assertEqual(len(analytics.permutations(100), 100)
```

Comment:
Missing closing paren at the end of this line.

`self.assertEqual(len(analytics.permutations(100), 100))`